.github/workflows/hrm-ci.yml

```diff
@@ -54,7 +54,7 @@ jobs:
           file: ./jobs/Dockerfile
           build-args: |
             lambda_name=${{ matrix.lambda_name }}
-            lambda_dir=${{ (startsWith(matrix.lambda_name, 'resources') && 'resources') || 'hrm' }}-domain
+            lambda_dir=${{ (startsWith(matrix.lambda_name, 'job') && 'jobs') || (startsWith(matrix.lambda_name, 'resources') && 'resources-domain') || 'hrm-domain' }}
```

Comment:
This is getting a little hairy now, can't see us needing to change it again for a while but should do something better next time we do

Reply:
This entire workflow probably needs an overhaul soon. The complexity is growing significantly with some of the new functionality, the repo reorg, and the quickly growing list of async-jobs.

I have ideas around simplifying the environment-region-map.json and making this a bit more dynamic now that I figured out some work-arounds. Once the lambda deployment side of things can be dynamically based on tags, it may make sense to move this to a "deploy all lambdas" by default with a "type your domain and your lambda name" option. As the list of lambdas grows, it is going to become more and more fragile to deploy a single lambda at a time. It would probably require a bit more repo reorg. Specifically creating domain specific jobs folders to contain all of the lambdas so it isn't mixed in with the rest service to make figuring out what is what a bit easier)

For more dynamic lambda deployment In the e2e deploy I moved to [name based parsing](https://github.com/techmatters/flex-plugins/blob/master/.github/workflows/e2e_lambda_deploy.yml#L116), but it is a bit clunky and fragile.

Using tags on the lambda is probably the right way to figure out what to deploy, but `aws lambda list-function` doesn't return tags or support any tag based filters unfortunately. I found this work around which will let us get lambda arns based on tag filters:

```
aws resourcegroupstaggingapi get-resources --resource-type-filters lambda --tag-filters Key=env,Values=development
```

The next iteration should probably set a tag with the ecr image identifier so that we can grab ARNs based on that tag and the environment tag to deploy to so we can perform that action dynamically instead of having to hard code it in the env/region map.